### PR TITLE
Fix/3216

### LINKED ISSUE #3216: stacks-node freezes periodically when syncing from genesis
**Describe the bug**
When syncing from genesis, the stacks-node itself would inconsistently freeze for hours at STX blocks above 45000+, not printing a single log. Sometimes it would unfreeze hours later and continue, other times it would not. After restarting the stacks-node, it would churn through 1 or 2 burnblocks, then freeze again. After restarting the node a few times and getting past a handful of burn blocks, it would continue for some time before freezing again.

stacks-blockchain node version: `2.05.0.2.2`

**Steps To Reproduce**
1. Start a stacks-node from genesis
2. Sync to chain tip, tracking `stacks_tip_height`.
3. Watch for times when `stacks_tip_height` hasn't increased for a while, and logs from the stacks-node have fully ceased despite the process is still running.

**Expected behavior**
The stacks-node should not freeze at any time when syncing.

Logs from stacks-node right before freezing
```
{"msg":"Could not log transaction receive to process time, txid not found in mempool","level":"DEBG","ts":"2022-07-23T18:18:05.037848600+00:00","thread":"chains-coordinator","line":224,"file":"src/monitoring/mod.rs","txid":"1e84ab42ba093d22faba70cf8ec66532a9ff633b2fdce5f108be769a54dc1f96"}
{"msg":"Reached chain tip df0ba246500b334458de6f6da0cedb6937b2cfbf/3347e502d26aaa7996e7ff8a6f1ee5f236cffce7b25ab9424b14f1a8378f4124 from c37372138c3c8d6b116e8c54fb09ab6336c1b036/4f43ae35c073ecb2efcd380916862032996aeda4a03d8d608b6ac829c765f82b","level":"DEBG","ts":"2022-07-23T18:18:05.038173200+00:00","thread":"chains-coordinator","line":5869,"file":"src/chainstate/stacks/db/blocks.rs"}
{"msg":"Event dispatcher: Successful POST","level":"DEBG","ts":"2022-07-23T18:18:06.391452905+00:00","thread":"chains-coordinator","line":139,"file":"testnet/stacks-node/src/event_dispatcher.rs","url":"http://stacks-node-api-snapshot-0.stacks-node-api-snapshot.prod-stacks.svc.cluster.local:3700/new_block"}
{"msg":"Accept block df0ba246500b334458de6f6da0cedb6937b2cfbf/3347e502d26aaa7996e7ff8a6f1ee5f236cffce7b25ab9424b14f1a8378f4124 as 8625909029193c3f947f737a26fd11e810dd3026ee8d1886ad50bdc09693b4b3","level":"DEBG","ts":"2022-07-23T18:18:06.392149260+00:00","thread":"chains-coordinator","line":2529,"file":"src/chainstate/stacks/db/blocks.rs"}
{"msg":"Accepted Stacks block df0ba246500b334458de6f6da0cedb6937b2cfbf/3347e502d26aaa7996e7ff8a6f1ee5f236cffce7b25ab9424b14f1a8378f4124 builds on the memoized canonical chain tip (00000000000000000007385e983a819aa55d96b76560fb06b66fe2f28d2c6d26)","level":"DEBG","ts":"2022-07-23T18:18:06.395800323+00:00","thread":"chains-coordinator","line":1391,"file":"src/chainstate/burn/db/sortdb.rs"}
{"msg":"Committing Clarity block connection","level":"DEBG","ts":"2022-07-23T18:18:06.397667280+00:00","thread":"chains-coordinator","line":568,"file":"src/clarity_vm/clarity.rs","index_block":"8625909029193c3f947f737a26fd11e810dd3026ee8d1886ad50bdc09693b4b3"}
{"msg":"commit_to(8625909029193c3f947f737a26fd11e810dd3026ee8d1886ad50bdc09693b4b3)","level":"DEBG","ts":"2022-07-23T18:18:06.397781151+00:00","thread":"chains-coordinator","line":460,"file":"src/clarity_vm/database/marf.rs"}
{"msg":"Dump already-sealed trie for b40ffa12cd6cdec17a0cf0231bffb40d9271c319a5113e88792a3488765e6c6b (root hash was 80d5471e490c0648d0bc73ff4e8db03e5a87753ebfc8a1a99bc247581755a1a5)","level":"DEBG","ts":"2022-07-23T18:18:06.416935298+00:00","thread":"chains-coordinator","line":370,"file":"src/chainstate/stacks/index/storage.rs"}
{"msg":"Flush: b40ffa12cd6cdec17a0cf0231bffb40d9271c319a5113e88792a3488765e6c6b to 8625909029193c3f947f737a26fd11e810dd3026ee8d1886ad50bdc09693b4b3","level":"DEBG","ts":"2022-07-23T18:18:06.418424740+00:00","thread":"chains-coordinator","line":1681,"file":"src/chainstate/stacks/index/storage.rs"}
{"msg":"Retarget block b40ffa12cd6cdec17a0cf0231bffb40d9271c319a5113e88792a3488765e6c6b to 8625909029193c3f947f737a26fd11e810dd3026ee8d1886ad50bdc09693b4b3","level":"DEBG","ts":"2022-07-23T18:18:06.418509706+00:00","thread":"chains-coordinator","line":1709,"file":"src/chainstate/stacks/index/storage.rs"}
{"msg":"Wrote block trie 8625909029193c3f947f737a26fd11e810dd3026ee8d1886ad50bdc09693b4b3 to rowid 62532 offset 14165540188","level":"DEBG","ts":"2022-07-23T18:18:06.424420415+00:00","thread":"chains-coordinator","line":315,"file":"src/chainstate/stacks/index/trie_sql.rs"}
{"msg":"Flush: identifier of 8625909029193c3f947f737a26fd11e810dd3026ee8d1886ad50bdc09693b4b3 is 62532","level":"DEBG","ts":"2022-07-23T18:18:06.424561302+00:00","thread":"chains-coordinator","line":1737,"file":"src/chainstate/stacks/index/storage.rs"}
{"msg":"Indexed-commit: MARF index","level":"DEBG","ts":"2022-07-23T18:18:06.425675762+00:00","thread":"chains-coordinator","line":887,"file":"src/util_lib/db.rs"}
{"msg":"Dump already-sealed trie for 8625909029193c3f947f737a26fd11e810dd3026ee8d1886ad50bdc09693b4b3 (root hash was 1d28b2eac5b7f3a9c7b826bcf6f3483b2ce0b02308bc988ffddb2881adad1b45)","level":"DEBG","ts":"2022-07-23T18:18:06.425767690+00:00","thread":"chains-coordinator","line":370,"file":"src/chainstate/stacks/index/storage.rs"}
{"msg":"Flush: 8625909029193c3f947f737a26fd11e810dd3026ee8d1886ad50bdc09693b4b3 to self","level":"DEBG","ts":"2022-07-23T18:18:06.426003480+00:00","thread":"chains-coordinator","line":1681,"file":"src/chainstate/stacks/index/storage.rs"}
{"msg":"Wrote block trie 8625909029193c3f947f737a26fd11e810dd3026ee8d1886ad50bdc09693b4b3 to rowid 51876 offset 769678341","level":"DEBG","ts":"2022-07-23T18:18:06.429201316+00:00","thread":"chains-coordinator","line":315,"file":"src/chainstate/stacks/index/trie_sql.rs"}
{"msg":"Flush: identifier of self is 51876","level":"DEBG","ts":"2022-07-23T18:18:06.429325111+00:00","thread":"chains-coordinator","line":1737,"file":"src/chainstate/stacks/index/storage.rs"}
{"msg":"Indexed-commit: MARF index","level":"DEBG","ts":"2022-07-23T18:18:06.429950131+00:00","thread":"chains-coordinator","line":887,"file":"src/util_lib/db.rs"}
{"msg":"Bump blocks processed","level":"DEBG","ts":"2022-07-23T18:18:06.432093472+00:00","thread":"chains-coordinator","line":733,"file":"src/chainstate/coordinator/mod.rs"}
{"msg":"Atlas: 1 attachment instances emitted from events","level":"INFO","ts":"2022-07-23T18:18:06.432265031+00:00","thread":"chains-coordinator","line":774,"file":"src/chainstate/coordinator/mod.rs"}

# No new logs printed after this point, and the node had to be restarted. However, this issue continued to happen afterwards
```

### PR DESCRIPTION
**Pull request by jcnelson**
This fixes #3216 by spawning the p2p thread _before_ processing an unbound number of sortitions.  The tricky part is that if there are no sortitions processed, then the node waits for the first sortition to be processed before starting the p2p thread.

### CODE REVIEW

File: testnet/stacks-node/src/run_loop/neon.rs [hunk 1]
```diff
@@ -542,33 +550,47 @@ impl RunLoop {
...
-            .expect("Unable to get burnchain tip");
+        // Make sure at least one sortition has happened
+        let sortdb = burnchain.sortdb_mut();
+        let (rc_aligned_height, sn) =
+            RunLoop::get_reward_cycle_sortition_db_height(&sortdb, &burnchain_config);
+
+        let burnchain_tip_snapshot = if sn.block_height == burnchain_config.first_block_height {
+            // need at least one sortition to happen.
+            burnchain
+                .wait_for_sortitions(Some(1))
```
Comment (kantai) [inline testnet/stacks-node/src/run_loop/neon.rs:568]: I think `wait_for_sortitions` uses real block height, not "sortition height" for its argument.

Comment (jcnelson) [reply testnet/stacks-node/src/run_loop/neon.rs]: Yup; fixed

File: testnet/stacks-node/src/run_loop/neon.rs [hunk 2]
```diff
@@ -542,33 +550,47 @@ impl RunLoop {
...
             coordinator_senders.clone(),
             attachments_rx,
         );
-        let sortdb = burnchain.sortdb_mut();
-        let mut sortition_db_height = RunLoop::get_sortition_db_height(&sortdb, &burnchain_config);
+
+        // Wait for all pending sortitions to process
+        let mut burnchain_tip = burnchain
+            .wait_for_sortitions(None)
+            .expect("Unable to get burnchain tip");
```
Comment (kantai) [inline testnet/stacks-node/src/run_loop/neon.rs:587]: Do you still think we should do this?

Comment (jcnelson) [reply testnet/stacks-node/src/run_loop/neon.rs:587]: Yes. This runs once all the rest of the threads are booted up and running, so the net effect here is that any unprocessed sortitions get processed before the node downloads more Bitcoin blocks.

Comment (kantai) [reply testnet/stacks-node/src/run_loop/neon.rs:587]: Okay, works for me

**Review by kantai — APPROVED** [review]
This LGTM, this just needs a changelog entry

**Review by gregorycoppola — APPROVED** [review]

**Pull request merged**